app/src/main/java/com/framgia/wsm/screen/confirmrequestleave/ConfirmRequestLeaveViewModel.java

```diff
@@ -1,98 +1,120 @@
+                RequestLeaveViewModel.LeaveType.IN_LATE_M) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.IN_LATE_WOMAN_A) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.IN_LATE_WOMAN_M) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.FORGOT_CARD_IN) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.FORGOT_TO_CHECK_IN) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.FORGOT_CARD_ALL_DAY) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.FORGOT_CHECK_ALL_DAY) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.LEAVE_OUT);
     }
 
-    public void setVisibleLayoutCheckout(boolean visibleLayoutCheckout) {
-        mIsVisibleLayoutCheckout = visibleLayoutCheckout;
-        notifyPropertyChanged(BR.visibleLayoutCheckout);
+    public boolean isVisibleLayoutCompensation() {
+        return mLeaveType.equals(RequestLeaveViewModel.LeaveType.IN_LATE_A) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.IN_LATE_M) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.LEAVE_EARLY_A) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.LEAVE_EARLY_M) || mLeaveType.equals(
+                RequestLeaveViewModel.LeaveType.LEAVE_OUT);
     }
 
-    @Bindable
-    public boolean isVisibleLayoutCheckin() {
-        return mIsVisibleLayoutCheckin;
+    @Override
+    public void onStart() {
+        mPresenter.onStart();
     }
 
     @Override
-    public void setVisibleLayoutCheckin(boolean isVisible) {
-        mIsVisibleLayoutCheckin = isVisible;
-        notifyPropertyChanged(BR.visibleLayoutCheckin);
+    public void onStop() {
+        mPresenter.onStop();
     }
 
-    @Bindable
-    public boolean isVisibleLayoutCompensation() {
-        return mIsVisibleLayoutCompensation;
+    @Override
+    public void onCreateFormRequestLeaveSuccess() {
+        mNavigator.startActivityAtRoot(MainActivity.class);
     }
 
     @Override
-    public void setVisibleLayoutCompensation(boolean visibleLayoutCompensation) {
-        mIsVisibleLayoutCompensation = visibleLayoutCompensation;
-        notifyPropertyChanged(BR.visibleLayoutCompensation);
+    public void onCreateFormFormRequestLeaveError(BaseException exception) {
+        mDialogManager.dialogError(exception);
     }
 
-    @Bindable
-    public String getLeaveTypeName() {
-        return mLeaveTypeName;
+    @Override
+    public void onGetUserSuccess(User user) {
+        if (user == null) {
+            return;
+        }
+        mUser = user;
+        notifyPropertyChanged(BR.user);
     }
 
-    private void setLeaveTypeName(String leaveTypeName) {
-        mLeaveTypeName = leaveTypeName;
-        notifyPropertyChanged(BR.leaveTypeName);
+    @Override
+    public void onGetUserError(BaseException exception) {
+        Log.e(TAG, "ConfirmRequestLeaveViewModel", exception);
```

Comment:
ERROR: [LongLogTag] The logging tag can be at most 23 characters, was 28 (ConfirmRequestLeaveViewModel)
 Log tags are only allowed to be at most 23 tag characters long.